src/psyclone/tests/psyir/transformations/intrinsics/matmul2code_trans_test.py

```diff
@@ -507,64 +531,70 @@ def test_validate11():
     matrix.children[2] = my_range
     with pytest.raises(TransformationError) as excinfo:
         trans.validate(matmul)
-    assert ("To use matmul2code_trans on matmul, only the first two indices "
-            "of the 1st argument are permitted to be Ranges but "
-            "found Range at index 2." in str(excinfo.value))
+    assert ("Transformation Error: To use matmul2code_trans on matmul, "
+            "each Range index of the argument 'x' must be a full range "
+            "but found non full range at position 2." in str(excinfo.value))
 
 
-def test_validate12():
+def test_validate13():
     '''Check that the Matmul2Code validate method raises the expected
     exception when the supplied node is a MATMUL IntrinsicCall but
-    the first dimension of its second (vector) argument is not a full
-    range.
+    a dimension of the second (vector) argument is indexed via a partial range.
 
     '''
     trans = Matmul2CodeTrans()
     matmul = create_matmul()
     vector = matmul.arguments[1]
-    vector.children[0] = Literal("1", INTEGER_TYPE)
+    my_range = vector.children[0].copy()
+    vector.children[2] = my_range
     with pytest.raises(TransformationError) as excinfo:
         trans.validate(matmul)
-    assert ("To use matmul2code_trans on matmul, the first index of the 2nd "
-            "argument 'y' must be a full range." in str(excinfo.value))
+    assert ("Transformation Error: To use matmul2code_trans on matmul, "
+            "each Range index of the argument 'y' must be a full range "
+            "but found non full range at position 2." in str(excinfo.value))
 
 
-def test_validate_2nd_dim_2nd_arg():
-    ''' Check that the Matmul2Code validate method raises the expected
-    exception when the second dimension of the second argument to MATMUL
-    is not a full range. '''
+def test_validate14():
+    '''
+    Check that the Matmul2Code validate method raises the expected
+    exception when the supplied node is a MATMUL IntrinsicCall but
+    more than two full ranges are specified in the first (matrix) argument.
+
+    '''
     trans = Matmul2CodeTrans()
     matmul = create_matmul()
-    matrix2 = matmul.arguments[1]
-    matrix2.children[1] = Range.create(Literal("1", INTEGER_TYPE),
-                                       Literal("2", INTEGER_TYPE))
+    matrix = matmul.arguments[0]
+    matrix.children[2] = Range.create(Literal("1", INTEGER_TYPE),
+                                      Literal("15", INTEGER_TYPE))
     with pytest.raises(TransformationError) as excinfo:
         trans.validate(matmul)
-    assert ("To use matmul2code_trans on matmul for a matrix-matrix "
-            "multiplication, the second index of the 2nd argument 'y' must "
-            "be a full range." in str(excinfo.value))
+    assert ("Transformation Error: To use matmul2code_trans on matmul, "
+            "no more than two indices of the argument 'x' "
```

Comment:
`debug_string()` again.

Reply:
Addressed in 10d5abbfb758dbaa45439829f974a99504e27227.

Reply:
Ah, I see now that I might have confused you here - apologies. I meant 'use `debug_string()` when creating the error message', not when testing it!